include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp

```diff
@@ -173,48 +229,68 @@ namespace SuperiorMySqlpp
             binding.length = &blob.counterRef();
         }
 
+        /**
+         * Result binding initialization specialization for types bindable into SQL type Date.
+         */
         template<typename T, typename std::enable_if<CanBindAsResult<BindingTypes::Date, T>::value, int>::type=0>
         inline void initializeResultBinding(MYSQL_BIND& binding, T& date)
         {
             binding.buffer = &date.detail_getBufferRef();
             binding.buffer_type = detail::toMysqlEnum(FieldTypes::Date);
         }
 
+        /**
+         * Result binding initialization specialization for types bindable into SQL type Time.
+         */
         template<typename T, typename std::enable_if<CanBindAsResult<BindingTypes::Time, T>::value, int>::type=0>
         inline void initializeResultBinding(MYSQL_BIND& binding, T& time)
         {
             binding.buffer = &time.detail_getBufferRef();
             binding.buffer_type = detail::toMysqlEnum(FieldTypes::Time);
         }
 
+        /**
+         * Result binding initialization specialization for types bindable into SQL type Datetime.
+         */
         template<typename T, typename std::enable_if<CanBindAsResult<BindingTypes::Datetime, T>::value, int>::type=0>
         inline void initializeResultBinding(MYSQL_BIND& binding, T& datetime)
         {
             binding.buffer = &datetime.detail_getBufferRef();
             binding.buffer_type = detail::toMysqlEnum(FieldTypes::Datetime);
         }
 
+        /**
+         * Result binding initialization specialization for types bindable into SQL type Timestamp.
+         */
         template<typename T, typename std::enable_if<CanBindAsResult<BindingTypes::Timestamp, T>::value, int>::type=0>
         inline void initializeResultBinding(MYSQL_BIND& binding, T& timestamp)
         {
             binding.buffer = &timestamp.detail_getBufferRef();
             binding.buffer_type = detail::toMysqlEnum(FieldTypes::Timestamp);
         }
 
+        /**
+         * Result binding initialization specialization for types bindable into SuperiorMySqlpp::Nullable.
+         */
         template<typename T, typename std::enable_if<CanBindAsResult<BindingTypes::Nullable, T>::value, int>::type=0>
         inline void initializeResultBinding(MYSQL_BIND& binding, T& nullable)
         {
             initializeNullable(binding, nullable);
+            // Note that initialization of payload is here unconditional, unlike in initializeParamBinding version
             initializeResultBinding(binding, *nullable);
         }
 
 
-        /*
-         * Initialize binding implementation
+        /**
+         * Initializes individual binding.
+         * Generic template, actual implementation is in specializations.
          */
         template<bool isParamBinding>
         struct InitializeBindingImpl;
 
+        /**
+         * Specialization of #InitializeBindingImpl for query parameters.
```

Comment:
Explain please the specialization by true/false template argument

Reply:
I added this information to the general version.